internal/controller/postgrescluster/util.go

```diff
@@ -285,3 +287,61 @@ func safeHash32(content func(w io.Writer) error) (string, error) {
 	}
 	return rand.SafeEncodeString(fmt.Sprint(hash.Sum32())), nil
 }
+
+// AdditionalVolumeMount returns the name and mount path of the additional volume.
+func AdditionalVolumeMount(volumeName, claimName string) corev1.VolumeMount {
+	return corev1.VolumeMount{
+		Name:      claimName,
+		MountPath: "/volumes" + "/" + volumeName}
+}
+
+// addAdditionalVolumesToSpecifiedContainers adds additional volumes to the specified
+// containers in the specified pod
+func addAdditionalVolumesToSpecifiedContainers(template *corev1.PodTemplateSpec,
+	additionalVolumes []*v1beta1.AdditionalVolume) {
+
+	for _, additionalVolumeRequest := range additionalVolumes {
+
+		additionalVolumeMount := AdditionalVolumeMount(
+			additionalVolumeRequest.Name,
+			additionalVolumeRequest.ClaimName)
+
+		additionalVolume := corev1.Volume{
+			Name: additionalVolumeMount.Name,
+			VolumeSource: corev1.VolumeSource{
+				PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+					// TODO: Using the claim name as the claim name here means this has to be unique
+					// We could add the index if we want someone to mount the same PVC in multiple dirs?
+					ClaimName: additionalVolumeMount.Name,
+					ReadOnly:  additionalVolumeRequest.ReadOnly,
```

Comment:
❓ I'm getting a little mixed up with which name is where. Should this be `additionalVolumeRequest.claimName` ?

Reply:
This is a little confusing because I'm switching from using the derived Mount spec to our AdditionalVolume spec -- I think I can clean this up by (a) using the request name above, (b) creating the Mount spec with the ReadOnly field, and (c) sticking to the Mount spec here.

Reply:
Eh, tackling this and the comment re: naming the volumes, the code is less intuitive than I wanted